### PR TITLE
Enable mpi_mod_int test case to take full-range MPI integers

### PR DESCRIPTION
**Pull request by tom-cosgrove-arm**
Also add commented-out test cases that currently fail

## Gatekeeper checklist

- [x] **changelog** not required - this is tests
- [x] **backport** #6575  
- [ ] **tests** these are tests

### CODE REVIEW

**Review by wernerlewis — DISMISSED** [review]
LGTM

**Review by gilles-peskine-arm — CHANGES_REQUESTED** [review]
The test code is using the wrong type for signed integers. And while you're at it please make a couple of improvements in the data.

File: tests/suites/test_suite_bignum.function [hunk 1]
```diff
@@ -981,11 +981,21 @@ void mpi_mod_int( char * input_X, char * input_Y,
     TEST_EQUAL( Y.n, 1 );
     TEST_EQUAL( A.n, 1 );
 
-    /* Convert the MPIs for Y and A to signed mbedtls_mpi_uints */
-    mbedtls_mpi_uint y = Y.p[0];
+    /* Convert the MPIs for Y and A to (signed) mbedtls_mpi_sints */
+
+    /* Since we're converting sign+magnitude to two's complement, we lose one
```
Comment (gilles-peskine-arm) [inline tests/suites/test_suite_bignum.function:986]: Technically, we don't require `mbedtls_mpi_sint` to have two's complement with naive overflow on conversion. If the value is out of range, the behavior is undefined, and could for example be a runtime fatal error (or nasal demons). But it's ok, we don't need to be robust against malformed test data. There is already plenty of test code that just assumes test data is within some allowed range.

**Review by gilles-peskine-arm — APPROVED** [review]
LGTM

**Review by wernerlewis — APPROVED** [review]

**Comment by yanesca** [comment]
(Internal CI has passed)

**Pull request merged**